### PR TITLE
feat: add support of gitlab groups as gitlab source

### LINKED ISSUE #422: Feature Request: Support of GitLab Groups in the GitLabSource (groupUrl)
**Problem**
When configuring a gitlab source right now it's only possible to use a project url which means that you can only configure webhooks for GitLab projects but not for GitLab groups.

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)**
Which persona is this feature for? -

**Exit Criteria**
Configure a group url in GitLab => The GitLab webhook was created for the GitLab group

**Time Estimate (optional):**
1-2

**Additional context (optional)**
It's seems that you just have to use a different GitLab API to setup the webhook in a GitLab group but technically it's seems like the webhook is exactly the same. I set the webhook manually for a GitLab group and it works just fine.

### PR DESCRIPTION
**Pull request by arkadiusjonczek**
Fixes #422

## Proposed Changes

- 🎁 Add support of gitlab groups as gitlab source

**Release Note**

```release-note
With this change it's also possible to use gitlab groups next to gitlab projects as event source using the `groupUrl`.

In the past it was only possible to use gitlab projects as event source using the `projectUrl`.

Backwards compatibility was an important requirement so if you only want to continue to use gitlab projects using the `projectUrl` you have to do nothing since there are no breaking changes.
```

---

Note: The `CustomResourceDefinition` will be updated asap

### CODE REVIEW

**Comment by Cali0707** [comment]
Thanks for working on this @arkadiusjonczek !!

/ok-to-test

I'll take a closer look next week, but let's run the CI for now

**Comment by arkadiusjonczek** [comment]
Hi @Cali0707 , thanks for running the CI.

I added the missing "Go license boilerplate" to the `webhook_client_test.go` file referenced in https://github.com/knative-extensions/eventing-gitlab/actions/runs/7154010905/job/19688338096?pr=431

**Review by Cali0707 — CHANGES_REQUESTED** [review]

File: pkg/apis/sources/v1alpha1/gitlabsource_types.go
Comment (arkadiusjonczek) [reply pkg/apis/sources/v1alpha1/gitlabsource_types.go:59]: I get your point. My initial idea was to have both properties and use `anyOf` in the CRD to prevent setting both.

Switching to `GitlabURL` would result in breaking changes what I wanted to prevent. If this is not a problem I'm fine with your approach.

Comment (Cali0707) [reply pkg/apis/sources/v1alpha1/gitlabsource_types.go:59]: Yeah good point on the no breaking changes if possible - I didn’t know about `anyOf`, very cool!

Maybe before going one route it’s worth checking in with @matzew over whether a change to the CRD (even just introducing the new field) requires bumping the CRD version?

Comment (Cali0707) [reply pkg/apis/sources/v1alpha1/gitlabsource_types.go:59]: Having thought about it more, let's go forwards with `anyOf` for now, and make the changes to the `v1beta1` CRD. If @matzew or Pierangelo (on vacation until the new year so I won't ping him here) think that these changes should be made to a `v1beta2` CRD it should be pretty straightforward to make that change if needed.

Thanks for all your work on this @arkadiusjonczek !!

Comment (arkadiusjonczek) [reply pkg/apis/sources/v1alpha1/gitlabsource_types.go:59]: Okay great to hear that. So whats the next steps here?

Related to https://github.com/knative-extensions/eventing-gitlab/blob/main/config/300-gitlabsource.yaml#L78 the current CRD version is `v1alpha1`. So you want to update this from `v1alpha1` to `v1beta1` right?

Comment (Cali0707) [reply pkg/apis/sources/v1alpha1/gitlabsource_types.go:59]: Ah if it's `v1alpha1` then we can leave it for now, I assumed it was `v1beta1`, my bad!

So next steps would be for you to update the CRD you linked above to have the `anyOf` validation, then I will take a second look at the PR.

Probably early in the new year another eventing approver will give you the second review needed for this to merge (I haven't done enough work in this repo to feel comfortable approving without a second look from someone else). 

Let me know if you have any questions, or if I can help in any way!

**Comment by Cali0707** [comment]
Ah sory @arkadiusjonczek looks like we didn't do a good job of following up on your PR here.

/cc @pierDipi @creydr 
Would you guys be able to take a look?

**Comment by Cali0707** [comment]
@arkadiusjonczek I don't have access to a gitlab instance where I could easily test this PR, would you be able to record a demo of this working?

**Pull request closed without merging**
